### PR TITLE
Lesson 6 mosut sorin

### PR DESCRIPTION
**Pull request by sorinmosut**

### CODE REVIEW

File: lesson_6/mosut_sorin_functions_homework.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+# import pakcage for Excel files
```
Comment (MihaelaGit) [inline lesson_6/mosut_sorin_functions_homework.py:1]: oare de ce ai utilizat pachetul de import Excel?

**Review by MihaelaGit — COMMENTED** [review]
Felicitari pt tema ! :)

File: lesson_6/mosut_sorin_functions_homework.py (hunk 1 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py:1]: Pt a transforma abrevierile tarilor in denumiri complete am folosit un document excel cu abrevieri.

Comment (MihaelaGit) [reply lesson_6/mosut_sorin_functions_homework.py:1]: multumesc

File: lesson_6/mosut_sorin_functions_homework.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            if str.isnumeric(cover_in[iter_0]):
+                cover += cover_in[iter_0]
+            else:
+                break
+        return cover
+
+# cover_value_int lamda function convert the 'coverage' value in integer
+
+cover_value_int = lambda arg_1, arg_2: int(check_cover_value(
+    str.strip(raw_data[arg_1][1][len(raw_data[1][1]) - 1 - arg_2])))
```
Comment (mariuspodean) [inline lesson_6/mosut_sorin_functions_homework.py]: I'm sure you can find a far simpler way to do this. Also, pay extra care to how you format your code. This code could have been formatted as:
```python
cover_value_int = (
    lambda arg_1, arg_2: int(
        check_cover_value(
            str.strip(
                raw_data[arg_1][1][len(raw_data[1][1]) - 1 - arg_2]
            )
        )
    )
```

File: lesson_6/mosut_sorin_functions_homework.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+def list_dict_year_cover(arg_iter_1):
+    return [
+             {
+                'year': str.strip(description[1]
+                                  [len(raw_data[1][1]) - 1 - iter_2]),
+                'coverage': cover_value_int(arg_iter_1, iter_2)
+              }
+             for iter_2 in range(len(raw_data[1][1]))
+            ]
```
Comment (mariuspodean) [inline lesson_6/mosut_sorin_functions_homework.py:83]:
```python
[
    {
        'year': str.strip(description[1][len(raw_data[1][1]) - 1 - iter_2]),
        'coverage': cover_value_int(arg_iter_1, iter_2)
    }
    for iter_2 in range(len(raw_data[1][1]))
]
```

File: lesson_6/mosut_sorin_functions_homework.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+"""
+dict_year_cover function creates a list of dicts with key = year and
+value = coverage for each year
+"""
+
+
+def list_dict_year_cover(arg_iter_1):
+    return [
+             {
+                'year': str.strip(description[1]
```
Comment (mariuspodean) [inline lesson_6/mosut_sorin_functions_homework.py]: This:
```str.strip(description[1][len(raw_data[1][1]) - 1 - iter_2])```
is extremely hard to read. I'm sure you can find a simpler way to write this

File: lesson_6/mosut_sorin_functions_homework.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+country_in = input('\nIntroduceti tara pentru care doriti'
+                   'sa obtineti valorile anuale ale acoperirii : ').title()
+
+
+def get_country_data(dataset, country_arg):
+    return {country_arg: [[data_country['year'], data_country['coverage']]
+                          for data_country in dataset[country_arg]
+                          ]
+            }
```
Comment (mariuspodean) [inline lesson_6/mosut_sorin_functions_homework.py]:
```python
{
    country_arg: [
        [data_country['year'], data_country['coverage']]
        for data_country in dataset[country_arg]
    ]
}
```

File: lesson_6/mosut_sorin_functions_homework.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+perform_average function , return the average of coverage in a period of time
+for a individual country
+"""
+
+
+def perform_average(*cover_values):
+    contor = 0
+    for values_dif_zero in cover_values[0]:
+        if values_dif_zero > 0:
+            contor += 1
```
Comment (mariuspodean) [inline lesson_6/mosut_sorin_functions_homework.py]: contor can be computed with a simple comprehension & and a builtin function ;)

File: lesson_6/mosut_sorin_functions_homework.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
+# import pakcage for Excel files
+import csv
+import pep8
+
+fchecker = pep8.Checker('functions_homework.py', show_source=True)
```
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py:7]: nice check, in the future pycharm can do this check for us 👍

File: lesson_6/mosut_sorin_functions_homework.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+file_errors = fchecker.check_all()
+
+print("Found %s errors (and warnings)" % file_errors)
```
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py]: Here an `if` statement will be nice to make sure we print only if there are errors 

```
if file_errors:
    print("Found %s errors (and warnings)" % file_errors)
```

File: lesson_6/mosut_sorin_functions_homework.py [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+
+# python object for Excel file
+
+cuntryes_abr_file = open('countryes_abrev.csv')
```
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py]: When we open a file we also need to close the file after we use the content

```
csv_reader_object = csv.reader(cuntryes_abr_file)
cuntryes_abr_file.close()
```

File: lesson_6/mosut_sorin_functions_homework.py [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 b']),
+]
+
+# check_cover function check each 'coverage' value and formates it
+
+
+def check_cover_value(cover_in):
+    cover = ''
+    if str.isnumeric(cover_in):
```
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py]: nice check, in this way we avoid errors from `int()` conversion

File: lesson_6/mosut_sorin_functions_homework.py [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+        return int(cover_in)
+    elif cover_in.find(':') != -1:
+        return 0
+    else:
```
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py]: what cases is this `else` dealing with ? 😄

File: lesson_6/mosut_sorin_functions_homework.py (hunk 2 above)
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py]: this lambda is a bit complicated. It is not obvious from reading the code what it dose. Can you rewrite it as a user defined function (with def)?

File: lesson_6/mosut_sorin_functions_homework.py [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+
+
+def list_dict_year_cover(arg_iter_1):
+    return [
+             {
+                'year': str.strip(description[1]
+                                  [len(raw_data[1][1]) - 1 - iter_2]),
+                'coverage': cover_value_int(arg_iter_1, iter_2)
+              }
+             for iter_2 in range(len(raw_data[1][1]))
```
Comment (Liitle) [inline lesson_6/mosut_sorin_functions_homework.py]: there are rare cases when we want to iterate using an index. For this, we have tow alternatives:
* `for item in raw_data[1][1]` for when we need to use just the value (this is usually the case)
* for index, value in enumerate(raw_data[1][1])` for when we need an index ( enumerate will return a tuple for each iteration (index_1, value_2) )

**Review by Liitle — COMMENTED** [review]
Nice job! I really like the code.

Try to replace all iterations that are using indices

**Comment by sorinmosut** [comment]
> 
> 
> Nice job! I really like the code.
> 
> Try to replace all iterations that are using indices

Thanx for reviewing my code and thanx for nice word.
About the indices iterations i belive i have a "C++ defect" sice i was learning coding 20 yers ago , but i will adapt to Python quickly.

File: lesson_6/mosut_sorin_functions_homework.py (hunk 7 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py:7]: I have experienced something about clean coding in Pycharm but i need more practice, but it helps to see what was wrong and modifying by myself than Pycharm clean my code , now when i am learning coding.

File: lesson_6/mosut_sorin_functions_homework.py (hunk 8 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py]: done , thanx

File: lesson_6/mosut_sorin_functions_homework.py (hunk 2 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py]: Only later i ve learned how to nest in writing for a easyer reading of the code and also i found out that the pep8 didn t like the lambda functions. i have rewrite the code an i have made a lot of changes.

File: lesson_6/mosut_sorin_functions_homework.py (hunk 4 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py]: i ve changed also this part.

File: lesson_6/mosut_sorin_functions_homework.py (hunk 12 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py]: also this ...

File: lesson_6/mosut_sorin_functions_homework.py (hunk 6 above)
Comment (sorinmosut) [reply lesson_6/mosut_sorin_functions_homework.py]: and this ...

**Pull request closed without merging**